CMakeLists.txt

```diff
@@ -91,37 +100,64 @@ else()
     endif()
 endif()
 
+# define sources and most of install targets
+
+## extension modules
+file(GLOB GEOCODERS_SRC geocoders/*.json geocoders/*.md geocoders/*.py)
+file(GLOB GEOCODERS_EXTRA LIST_DIRECTORIES false geocoders/test/*)
+install(FILES ${GEOCODERS_SRC} DESTINATION ${DATADIR}/geocoders)
+
+file(GLOB GUIDES_SRC guides/*.json guides/*.md guides/*.qml guides/*.py)
+file(GLOB GUIDES_EXTRA LIST_DIRECTORIES false guides/test/*)
+install(FILES ${GUIDES_SRC} DESTINATION ${DATADIR}/guides)
+
+file(GLOB MAPS_SRC maps/*.json maps/*.md)
+install(FILES ${MAPS_SRC} DESTINATION ${DATADIR}/maps)
+
+file(GLOB ROUTERS_SRC routers/*.json routers/*.md routers/*.qml routers/*.py)
+file(GLOB ROUTERS_EXTRA LIST_DIRECTORIES false routers/test/*)
+install(FILES ${ROUTERS_SRC} DESTINATION ${DATADIR}/routers)
+
+## python sources: installed in subdir
+file(GLOB POOR_SRC poor/*.py)
```

Comment:
Why do you define this here rather than in `poor/CMakeLists.txt`?

Reply:
This is to make the sources all appear as a tree with the branches under it:

- Sources
  - qml
    - qml_file
  - src
  - poor

If I define all these sources under subfolders, I will get

- pure-maps
  - poor
     - poor-sources
  - qml
     - qml-sources
        - qml_file

In essence, it makes the tree organized as it is in the folder and allows me to avoid extra clicks to open each source file. So, while CMakeLists.txt is not as compartmentalized, developers can work more efficiently. Which is way more important than compartmentalized CMakeLists.